### PR TITLE
build: add a release mechanism workflow

### PR DESCRIPTION
**Pull request by YossiSaadi**
https://monday.monday.com/boards/3532714909/pulses/5702184212

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version-file: ".nvmrc"
+          registry-url: https://registry.npmjs.org/
+      - run: npm ci
+      - run: npm run lint
+      - run: npm run build
+      - name: Semantic Release
```
Comment (talkor) [inline .github/workflows/release.yml:17]: Will it skip if previous steps fail?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        uses: cycjimmy/semantic-release-action@v4
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+          GIT_AUTHOR_NAME: "ci"
+          GIT_AUTHOR_EMAIL: "deploy@monday.com"
```
Comment (talkor) [inline .github/workflows/release.yml:23]: If it's open source maybe worth using a different email or something more specific to Vine?

File: .releaserc.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+  "branches": ["master"]
```
Comment (talkor) [inline .releaserc.json:2]: Didn't you change it to main?

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "@mondaydotcomorg/storybook-addon-playground",
-  "version": "0.0.1",
+  "name": "storybook-addon-playground",
+  "version": "0.0.0",
```
Comment (talkor) [inline package.json:3]: If you publish it this way it should be 0.0.1 at least

Comment (talkor) [reply package.json:3]: Or you expecting it to bump it in this pr?

File: package.json [hunk 5]
```diff
@@ -46,8 +46,8 @@
   "scripts": {
     "build": "rollup -c",
     "build:watch": "npm run build -- -w",
-    "lint": "eslint .",
-    "lint:fix": "eslint . --fix",
+    "lint": "eslint . --max-warnings 0",
```
Comment (talkor) [inline package.json:49]: Awesome

**Review by talkor — APPROVED** [review]
Awesome, see comments :)

File: .github/workflows/release.yml (hunk 1 above)
Comment (YossiSaadi) [reply .github/workflows/release.yml:17]: Lint and build should throw if failing, and same for ci (install)

File: .github/workflows/release.yml (hunk 2 above)
Comment (YossiSaadi) [reply .github/workflows/release.yml:23]: It is the same git user being used in the other libs, so I went with this convention

File: .releaserc.json (hunk 3 above)
Comment (YossiSaadi) [reply .releaserc.json:2]: Nope. It was created as main (PC) but I've changed it to master

File: package.json (hunk 4 above)
Comment (YossiSaadi) [reply package.json:3]: I want it to start at .0 and running this workflow once it's merged (I guess before that I will create a branch with dry_run to not create a publish, just to print what if.., and will fix if something needs a fix) will bump to 0.0.1 and commit it and release to NPM the 0.0.1

**Pull request merged**